odoo/fields.py

```diff
@@ -1157,103 +1159,111 @@ def __get__(self, record, owner):
             # process pending computations
             self.recompute(record)
 
-        try:
-            value = env.cache.get(record, self)
-
-        except KeyError:
-            # behavior in case of cache miss:
-            #
-            #   on a real record:
-            #       stored -> fetch from database (computation done above)
-            #       not stored and computed -> compute
-            #       not stored and not computed -> default
-            #
-            #   on a new record w/ origin:
-            #       stored and not (computed and readonly) -> fetch from origin
-            #       stored and computed and readonly -> compute
-            #       not stored and computed -> compute
-            #       not stored and not computed -> default
-            #
-            #   on a new record w/o origin:
-            #       stored and computed -> compute
-            #       stored and not computed -> new delegate or default
-            #       not stored and computed -> compute
-            #       not stored and not computed -> default
-            #
-            if self.store and record.id:
-                # real record: fetch from database
-                recs = record._in_cache_without(self)
+        value = env.cache.get(self, record.env.cache_key(self), record._ids[0])
+        if value is NOTHING or (self.translate and not self._get_translation_validator(self._lang(record.env))(value)):
+            value = self._get(record)
```

Comment:
Obviously you miss an abstraction here.  This special case for translated fields is inappropriate in this generic code:
```suggestion
        try:
            value = self.get_cache(record)
        except CacheMiss:
            value = self._get(record)
```
The validator for translations is managed by `_String.get_cache`.